src/Microsoft.Identity.Web/CertificateManagement/DefaultCertificateLoader.cs

```diff
@@ -203,7 +199,7 @@ private static void ParseStoreLocationAndName(string storeDescription, ref Store
             DefaultCertificateLoader defaultCertificateLoader = new DefaultCertificateLoader();
             CertificateDescription certDescription = certificateDescription.First();
             defaultCertificateLoader.LoadIfNeeded(certDescription);
-            return certDescription?.Certificate;
+            return certDescription.Certificate!;
```

Comment:
`Certificate` here, I assume, should never be null. But if it's not found in the cert store, it will be null. I think this will result in passing null to MSAL and to Wilson.

Reply:
MSAL will handle the cert being null and throw an `ArgumentNullException`, we catch the exception

---
In reply to: [444031657](https://github.com/AzureAD/microsoft-identity-web/pull/231#discussion_r444031657) [](ancestors = 444031657)

Reply:
- We do get an uncaught NPE when trying to set token decryption cert.
https://github.com/AzureAD/microsoft-identity-web/blob/6fe02dbb312bed2d7351ff9035a6e717cafc7574/src/Microsoft.Identity.Web/WebApiAuthenticationBuilderExtensions.cs#L117-L118

- Yes, we do catch the NPE from MSAL. But the exception is kind of general. Also, FromBase64 gives `Cannot find the requested object.`; FromDisk gives `The system cannot find the file specified.` Similar to my other comment, maybe we can update this to give better messaging and user experience when we implement cert rotation stuff.

Reply:
That's fine, but maybe something for a separate PR w/a new issue.